src/lib/swap/HtlcUtils.js

```diff
@@ -0,0 +1,168 @@
+/* global Nimiq */
+/* global BitcoinJS */
+/* global Errors */
+
+class HtlcUtils { // eslint-disable-line no-unused-vars
+    /**
+     * @param {unknown} data - Uint8Array
+     * @returns {NimHtlcContents}
+     */
+    static decodeNimHtlcData(data) {
+        const error = new Errors.InvalidRequestError('Invalid NIM HTLC data');
+
+        if (!data || !(data instanceof Uint8Array) || data.length !== 78) throw error;
+
+        const buf = new Nimiq.SerialBuffer(data);
+
+        const sender = Nimiq.Address.unserialize(buf).toUserFriendlyAddress();
+        const recipient = Nimiq.Address.unserialize(buf).toUserFriendlyAddress();
+        const hashAlgorithm = /** @type {Nimiq.Hash.Algorithm} */ (buf.readUint8());
+        const hashRoot = Nimiq.Hash.unserialize(buf, hashAlgorithm).toHex();
+        const hashCount = buf.readUint8();
+        const timeout = buf.readUint32();
+
+        if (!sender || !recipient || !hashRoot || !timeout) {
+            throw error;
+        }
+
+        if (hashAlgorithm !== Nimiq.Hash.Algorithm.SHA256) throw error;
+        if (hashCount !== 1) throw error;
+
+        return {
+            refundAddress: sender,
+            redeemAddress: recipient,
+            hash: hashRoot,
+            timeoutBlockHeight: timeout,
+        };
+    }
+
+    /**
+     * @param {unknown} script - Uint8Array
+     * @returns {BtcHtlcContents}
+     */
+    static decodeBtcHtlcScript(script) {
+        const error = new Errors.InvalidRequestError('Invalid BTC HTLC script');
+
+        if (!script || !(script instanceof Uint8Array) || !script.length) throw error;
+        const chunks = BitcoinJS.script.decompile(/** @type {Buffer} */ (script));
+        if (!chunks) throw error;
+        const asm = BitcoinJS.script.toASM(chunks).split(' ');
+
+        let branchesVerifiedIndividually = false;
```

Comment:
A BTC HTLC script can be one of two versions. One where both the redeem and refund branch is verified independently, and one where only one verification happens at the end. This method can parse both. (Should probably have tests...)

Reply:
Test are added in the `oasis` branch: b04b3648f9aaa6f34619a2fd5f5ccfe770db724b